pkg/kubernetes/status/readiness.go

```diff
@@ -0,0 +1,95 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	label "github.com/kudobuilder/kudo/pkg/util/kudo"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ComputeReadiness updates instance Status.Conditions.Ready based on current state of underlying resources
+// currently readiness examines the following types: Pods, StatefulSets, Deployments, ReplicaSets and DaemonSets
+// Instance is considered ready if all the resources linked to this instance are also ready
+func IsReady(i kudoapi.Instance, c client.Client) (ready bool, msg string, err error) {
```

Comment:
This class needs a unit test, with a bunch of preconfigured resources from a fake client ;)

Reply:
Yeah I was kind of expecting this feedback :D